contracts/token/ERC20/extensions/ERC20Crosschain.sol

```diff
@@ -0,0 +1,24 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.26;
+
+import {ERC20} from "../ERC20.sol";
+import {BridgeERC20} from "../../../crosschain/bridges/BridgeERC20.sol";
+
+abstract contract ERC20Crosschain is ERC20, BridgeERC20 {
+    /// @dev TransferFrom variant of {crosschainTransferFrom}, using ERC20 allowance from the sender to the caller.
+    function crosschainTransferFrom(address from, bytes memory to, uint256 amount) public virtual returns (bytes32) {
+        _spendAllowance(from, msg.sender, amount);
+        return _crosschainTransfer(from, to, amount);
+    }
```

Comment:
I think I'd remove this, to reduce surface, and because it's not standardized so I don't expect it to be used.

Reply:
its an easy work around, but even with multicall, that would still involves two storage update (for the balance of the intermediary account).